nemo/core/optim/lr_scheduler.py

```diff
@@ -501,33 +503,39 @@ def prepare_lr_scheduler(
     if 'max_steps' in scheduler_config and scheduler_config['max_steps'] is not None:
         max_steps = scheduler_config['max_steps']
 
-    elif 'iters_per_batch' in scheduler_config:
-        # Compute effective max_steps if iters_per_batch is provided
+    elif 't_max_epochs' in scheduler_config:
+        # Compute effective max_steps if t_max_epochs is provided
         if train_dataloader is None:
             logging.warning(
-                'As `iters_per_batch` is provided/computed, it is required to pass the train dataloader in order\n'
+                'As `t_max_epochs` is provided/computed, it is required to pass the train dataloader in order\n'
                 'to compute effective maximum number of steps.\n'
                 'Scheduler will not be instantiated !'
             )
             return None
 
-        # Raise exception if neither `max_steps` nor `iters_per_batch` is provided
-        if scheduler_config.get('iters_per_batch', None) is None:
+        # Raise exception if neither `max_steps` nor `t_max_epochs` is provided
+        if scheduler_config.get('t_max_epochs', None) is None:
             logging.warning(
-                "`iters_per_batch` cannot be None when `max_steps` is not not provided.\n"
+                "`t_max_epochs` cannot be None when `max_steps` is not not provided.\n"
                 "This can occur when `train dataloader` is not available to correctly "
                 "prepare the scheduler.\n"
                 "Scheduler will not be instantiated !"
             )
             return None
 
         # Get iters_per_batch
-        iters_per_batch = scheduler_config.get('iters_per_batch')
+        max_epochs = scheduler_config.get('t_max_epochs')
+        accumulate_grad_batches = scheduler_config.get('t_accumulate_grad_batches')
+        num_workers = scheduler_config.get('t_num_workers')
 
         # Compute effective num max_steps
         num_samples = len(train_dataloader.dataset)
         batch_size = train_dataloader.batch_size
-        max_steps = round(num_samples * iters_per_batch / float(batch_size))
+        drop_last = train_dataloader.batch_size
```

Comment:
Drop_last = train dataloader batch size ?

Reply:
That is a good catch